engine/apps/mattermost/tests/test_mattermost_channel.py

```diff
@@ -0,0 +1,381 @@
+
+    url = reverse("mattermost:channel-list")
+    with patch("apps.mattermost.client.requests.get", return_value=mock_response) as mock_request:
+        response = client.post(
+            url,
+            data={"team_name": "fuzzteam", "channel_name": "fuzzchannel"},
+            format="json",
+            **make_user_auth_headers(user, token),
+        )
+    mock_request.assert_called_once()
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert response.json()["detail"] == "Mattermost api call gateway timedout"
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "role,expected_status",
+    [
+        (LegacyAccessControlRole.ADMIN, status.HTTP_200_OK),
+        (LegacyAccessControlRole.EDITOR, status.HTTP_403_FORBIDDEN),
+        (LegacyAccessControlRole.VIEWER, status.HTTP_403_FORBIDDEN),
+        (LegacyAccessControlRole.NONE, status.HTTP_403_FORBIDDEN),
+    ],
+)
+def test_set_default_mattermost_channels_permissions(
+    make_organization_and_user_with_plugin_token,
+    make_user_auth_headers,
+    make_mattermost_channel,
+    role,
+    expected_status,
+):
+    client = APIClient()
+
+    organization, user, token = make_organization_and_user_with_plugin_token(role)
+    mattermost_channel = make_mattermost_channel(organization=organization)
+
+    url = reverse("mattermost:channel-set-default", kwargs={"pk": mattermost_channel.public_primary_key})
+    response = client.post(url, **make_user_auth_headers(user, token))
+
+    assert response.status_code == expected_status
+
+
+@pytest.mark.django_db
+def test_list_mattermost_channels(
+    make_organization_and_user_with_plugin_token, make_user_auth_headers, make_mattermost_channel
+):
+    client = APIClient()
+
+    organization, user, token = make_organization_and_user_with_plugin_token()
+
+    first_mattermost_channel = make_mattermost_channel(organization=organization)
+    second_mattermost_channel = make_mattermost_channel(organization=organization)
+
+    expected_payload = [
+        {
+            "id": first_mattermost_channel.public_primary_key,
+            "channel_id": first_mattermost_channel.channel_id,
+            "channel_name": first_mattermost_channel.channel_name,
+            "display_name": first_mattermost_channel.unique_display_name,
+            "is_default_channel": first_mattermost_channel.is_default_channel,
+        },
+        {
+            "id": second_mattermost_channel.public_primary_key,
+            "channel_id": second_mattermost_channel.channel_id,
+            "channel_name": second_mattermost_channel.channel_name,
+            "display_name": second_mattermost_channel.unique_display_name,
+            "is_default_channel": second_mattermost_channel.is_default_channel,
+        },
+    ]
+
+    url = reverse("mattermost:channel-list")
+    response = client.get(url, **make_user_auth_headers(user, token))
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json() == expected_payload
```

Comment:
```suggestion
    response_data = response.json()
    assert len(response_data) == 2
    for channel_data in expected_payload:
        assert channel_data in response_data
```
(you cannot guarantee an order in the response)